noxfile.py

```diff
@@ -232,8 +210,20 @@ def set_dev(session):
     set_environment_variables(PYBAMM_ENV, session=session)
     session.install("virtualenv", "cmake")
     session.run("virtualenv", os.fsdecode(VENV_DIR), silent=True)
-    python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
+    if sys.platform == "win32":
+        python = os.fsdecode(VENV_DIR.joinpath("Scripts\\python"))
+    else:
+        python = os.fsdecode(VENV_DIR.joinpath("bin/python"))
     components = ["all", "dev", "jax"]
+    with open("pyproject.toml", encoding="utf-8") as file:
+        toml_data = tomlkit.parse(file.read())
+    build_dependencies = toml_data.get("build-system").get("requires")
+    if sys.platform == "win32":
+        build_dependencies = [
+            dep
+            for dep in build_dependencies
+            if not (dep.startswith("casadi") or dep.startswith("cmake"))
+        ]
```

Comment:
Could you convert this to a reusable function at the top of the file and use it for all the sessions? All of the sessions should install the project without build isolation, not just `nox -s dev`. Also, we should drop `tomlkit` entirely from all workflows, see the `nox` documentation here on how to replace it –

https://github.com/wntrblm/nox/blob/e0868b6c092c80837f86463a54d34379851d43d7/docs/tutorial.rst#loading-dependencies-from-pyprojecttoml-or-scripts

Reply:
Okay, working on it!